main.py

```diff
@@ -4,13 +4,22 @@
 import urllib.request as request
 import sys
 
+regex_doi = r"doi\s?=\s?[{\"']([^}\"']+)[}\"']"
+regex_url = r"url\s?=\s?[{\"'](https?://doi\.org/([\w\.]+/[^}/\"']+))/?[}\"']"
+
 
 for line in fileinput.input():
     try:
-        if "doi={" in re.sub("\s+", "", line).lower():
-            doi = re.search("{(.+)}", line).group(1)
+        search_doi = re.search(regex_doi, re.sub("\s+", "", line).lower(), re.IGNORECASE)
```

Comment:
we don't need the `.lower` anymore, right?

Reply:
Good catch! Will fix that.